src/pages/posts/comparing-rust-smart-contract-storage-approaches.md

```diff
@@ -0,0 +1,515 @@
+
+```rust
+#[entry_point]
+pub fn execute(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Increment {} => {
+            STATE.update(deps.storage, |mut state| -> Result<_, ContractError> {
+                state.count += 1;
+                Ok(state)
+            })?;
+            Ok(Response::new())
+        }
+    }
+}
+
+#[entry_point]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::GetNum {} => {
+            let state = STATE.load(deps.storage)?;
+            let raw_response = GetNumResponse { count: state.count };
+            to_binary(&raw_response)
+        }
+    }
+}
+```
+
+
+<details>
+<summary>If you want a refresher on how those <code>ExecuteMsg</code> and <code>QueryMsg</code> enums were defined, scroll up or click here.</summary>
+
+```rust
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub count: i8,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    Increment {},
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    // GetNum returns the current count as a json-encoded number
+    #[returns(GetNumResponse)]
+    GetNum {},
+}
+
+// We define a custom struct for each query response
+#[cw_serde]
+pub struct GetNumResponse {
+    pub count: i8,
+}
+```
+</details>
+
+But really, this is just one way to read and write the data stored in that `STATE` item. `STATE` is a fully-typed Rust struct, with fantastic type-ahead [documentation](https://docs.rs/cw-storage-plus/1.0.0/cw_storage_plus/struct.Item.html) in your editor, so you can quickly figure out other ways to update state, if `update` doesn't fit your situation.
+
+# Complex data
+
+Now that we know our way around these different SDKs, let's do a quick side-by-side for how to store more realistic data. We'll look at how each network's standard Fungible Token contract stores its set of tokens. These will be simplified code samples to show the core storage operations.
+
+## NEAR
+
+`near-sdk-rs` provides [collections](https://docs.rs/near-sdk/latest/near_sdk/collections/index.html) and, more recently, [store](https://docs.rs/near-sdk/latest/near_sdk/store/index.html) modules with efficient data structures to store large sets of data in the key-value store. Most code in the wild, including [the example fungible token example](https://github.com/near/near-sdk-rs/tree/master/examples/fungible-token), still use `collections`.
+
+```rust
+use near_sdk::collections::LookupMap;
+
+// this comes from the standards library:
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct FungibleToken {
+    /// AccountID -> Account balance.
+    pub accounts: LookupMap<AccountId, Balance>,
+}
+
+// then, in your contract, this is your main struct:
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct Contract {
+    token: FungibleToken,
+}
+```
+
```

Comment:
I would add here how you initialize the map to make it clear that it uses a prefix.